nemo/collections/nlp/models/language_modeling/megatron_t5_prompt_learning_model.py

```diff
@@ -73,6 +74,7 @@ class MegatronT5PromptLearningModel(MegatronBasePromptLearningModel):
 
     def __init__(self, cfg: DictConfig, trainer: Trainer):
         super().__init__(cfg, trainer)
+        self.model_type = ModelType.encoder_and_decoder
```

Comment:
is this `self.model_type` used?

Reply:
Apex accesses this model property inside it's fwd/bwd func. I think all encoder-decoder models need this, but the default works fine for encoder only or decoder only models. Its needed to know if tensor shapes might be different between encoder and decoder communications when PP > 1 (PP > 2?)